commit ce4d8b16e64d062f1c6271a0068dbbbba2dee620
Author: Wenwen Wang <wenwen@cs.uga.edu>
Date:   Mon Aug 19 22:03:46 2019 -0500

    ubifs: Fix memory leak in read_znode() error path
    
    In read_znode(), the indexing node 'idx' is allocated by kmalloc().
    However, it is not deallocated in the following execution if
    ubifs_node_check_hash() fails, leading to a memory leak bug. To fix this
    issue, free 'idx' before returning the error.
    
    Fixes: 16a26b20d2af ("ubifs: authentication: Add hashes to index nodes")
    Signed-off-by: Wenwen Wang <wenwen@cs.uga.edu>
    Signed-off-by: Richard Weinberger <richard@nod.at>

diff --git a/fs/ubifs/tnc_misc.c b/fs/ubifs/tnc_misc.c
index 6f293f662d98..49cb34c3f324 100644
--- a/fs/ubifs/tnc_misc.c
+++ b/fs/ubifs/tnc_misc.c
@@ -284,6 +284,7 @@ static int read_znode(struct ubifs_info *c, struct ubifs_zbranch *zzbr,
 	err = ubifs_node_check_hash(c, idx, zzbr->hash);
 	if (err) {
 		ubifs_bad_hash(c, idx, zzbr->hash, lnum, offs);
+		kfree(idx);
 		return err;
 	}